src/tls_config.rs

```diff
@@ -0,0 +1,88 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use openssl::ssl::{ Ssl as SslImpl, SslContext, SslMethod, SSL_VERIFY_NONE };
+use openssl::ssl::error::SslError;
+use openssl::x509::X509FileType;
+use openssl_sys;
+
+use std::collections::HashMap;
+use std::io::{ Error, ErrorKind };
+use std::path::Path;
+
+pub struct TlsSniConfig {
+    ssl_hosts: HashMap<String, SslContext>
+}
+
+impl TlsSniConfig {
+
+    pub fn new() -> Self {
+        TlsSniConfig {
+            ssl_hosts: HashMap::new()
+        }
+    }
+
+    pub fn context(&self) -> Result<SslContext, Error> {
+        for ctx in self.ssl_hosts.values() {
+            return Ok(ctx.clone());
```

Comment:
I've added a comment.  tl;dr; Yes I wanted that.

The comment explains why this is :)